### PR TITLE
latest patches bad calc thou

### PR DESCRIPTION
**Pull request by WhyDoesGodDoThis**
naxstatus patched
snap yaw degrees patched
new align calc math

### CODE REVIEW

**Comment by dcowden** [comment]
@WhyDoesGodDoThis looks like this version has test failures.  I think it looks fine to merge once we fix that.

File: src/main/java/frc/robot/subsystems/NavxStatus.java [hunk 1]
```diff
@@ -21,9 +21,9 @@ public NavxStatus() {
     }
 
     public NavxStatus(double forward, double right, double yaw, double pitch) {
-        this.yawAngleDegrees = forward;
-        this.forwardDistance = right;
-        this.rightDistance = yaw;
+        this.yawAngleDegrees = yaw;
+        this.forwardDistance = forward;
+        this.rightDistance = right;
         this.pitchAngleDegrees = pitch;
```
Comment (mandrews281) [inline src/main/java/frc/robot/subsystems/NavxStatus.java:27]: Great catch.  That might explain why the NavX Pose positions didn't make sense while I was driving.  This was caused by the assignments being in a different order than the input parameters, so I'd move the this.yawAngleDegrees = yaw; after the assignment for rightDistance.

File: src/main/java/frc/robot/util/StoppingCounter.java [hunk 2]
```diff
@@ -14,14 +14,17 @@ public StoppingCounter(String name, int targetCounts) {
...
 			currentCounts = 0;
 		}
 		SmartDashboard.putNumber("StopCounter::" + name, currentCounts);
 		return currentCounts > targetCounts;
+	}
 
+	public void reset() {
+		this.currentCounts = 0;
 	}
 }
```
Comment (mandrews281) [inline src/main/java/frc/robot/util/StoppingCounter.java:30]: You need to add an initialize() method and zero your internal parameters in that method. 
I have a template installer (InstallEntechVSCodeMods.bat) that let's you use the right click New Class/Command menu.

**Review by mandrews281 — COMMENTED** [review]
Dave.  Since this also involves the Pose, I'd like you to look at it too.

**Pull request merged**